spec/unit/chunker_spec.rb

```diff
@@ -69,16 +96,45 @@ def @throttler.stride
                                                            :start     => 1,
                                                            :limit     => 1)
 
+      @connection.expects(:select_value).with(regexp_matches(/where id >= 1 order by id limit 1 offset 0/)).returns(nil)
```

Comment:
In this test the assumption is that the chunker should try to copy up to record number 10 (`:limit => 10`)

I made the assumption in the test that no records were added to the table while the LHM was running, so the max id for that table was still going to be 10 at the end of the LHM. 
In that case, a `select id .... where id >= 9 order by id limit 1 offset 2` was going to return no results. There are no ids at offset `2` of that recordset. Only at 0 (id = 9) and 1 (id = 10).

The test is actually going to work anyway by making that stub to return any value higher than 10 (for instance, if a thousand records were added to the table after the LHM started), since the upper limit for the chunker is still going to be calculated as `[top ? top.to_i : @limit, @limit].min`

Makes sense?